mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoDeleteMethodGenerator.java

```diff
@@ -141,16 +141,9 @@ public DaoDeleteMethodGenerator(
     }
 
     // Validate the return type:
-    DaoReturnType returnType = parseReturnType(methodElement.getReturnType());
-    Set<DaoReturnTypeKind> supportedReturnTypes = getSupportedReturnTypes();
-    if (!supportedReturnTypes.contains(returnType.getKind())) {
-      context
-          .getMessager()
-          .error(
-              methodElement,
-              "Invalid return type: %s methods must return one of %s",
-              Delete.class.getSimpleName(),
-              supportedReturnTypes);
+    DaoReturnType returnType =
+        parseAndValidateReturnType(getSupportedReturnTypes(), Delete.class.getSimpleName());
+    if (returnType == null) {
       return Optional.empty();
     }
```

Comment:
Not super happy about this. We could make `parseAndValidate...` return an optional, and map the rest of the method onto it. But then why not do it for everything else... I don't want to rewrite the whole method in a functional style.